portfolio/src/main/webapp/script.js

```diff
@@ -56,7 +57,84 @@ function getRandomFile(element, path, fileName, numElems) {
  * @param {DOM element} element - an element in the DOM for selecting
  */
 function fetchAndModify(path, element) {
-  fetch(path).then(response => response.text()).then(content => 
-    element.innerHTML= content
-  )
+  fetch(path)
+      .then(response => response.text())
+      .then(content => element.innerHTML = content)
 }
+/*
+ * This code creates an innerHTML and then reads its text to remove special
+ * characters from the input
+ * @param {string} str The input to remove special characters from to prevent
+ *     scripting attacks
+ */
+var sanitizeHTML = function(str) {
+  var temp = document.createElement('div');
+  temp.textContent = str;
+  return temp.innerHTML;
+};
+
+
+/*
+ * This function fetches the value return by the given data service and then
+ * modifies the content of a div to store each of the response values as a
+ * paragraph.
+ * @param {string} path The url of the data service to use
+ */
+function fetchAndChangeBody(path) {
+  fetch(path).then(response => response.json()).then(json => {
+    let index = 0;
+    for (index in json) {
+      console.log(json[index]);
+      const target = document.getElementById('comment');
+      let newComment = document.createElement('p');
+      newComment.innerText = json[index];
+      target.appendChild(newComment);
+      index++;
+    }
+  })
+}
+
+/*
+ * This function fetches the comment data and adds each of them to the loaded
+ * content.
+ * TODO(morleyd): change individual comment load to be asynchronous to improve
+ * performance. It might also make sense to have a hard limit on the number of
+ * comments loaded, think about this for future implementations.
+ */
+function loadComments() {
+  fetch('\data')
+      .then(response => response.json())
+      .then(
+          json => {
+            console.log(json)
+            var index = 0;
+            for (index in json) {
```

Comment:
Same recommendations for this loop: no need to predeclare, and use `key` rather than `index`.